code/modules/power/singularity/singularity.dm

```diff
@@ -358,17 +358,17 @@
 
 /obj/singularity/proc/toxmob()
 	var/toxrange = 10
-	var/toxdamage = 4
+	var/toxpwr = 4
 	var/radiation = 15
 	var/radiationmin = 3
-	if (src.energy>200)
-		toxdamage = round(((src.energy-150)/50)*4,1)
-		radiation = round(((src.energy-150)/50)*5,1)
-		radiationmin = round((radiation/5),1)//
+	if (energy>200)
+		toxpwr += round((energy-150)/30,1)
+		radiation += round((energy-150)/10,1)
+		radiationmin = round((radiation/5),1)
 	for(var/mob/living/M in view(toxrange, src.loc))
 		M.irradiate(rand(radiationmin,radiation))
-		toxdamage = (toxdamage - (toxdamage*M.getarmor(null, "rad")))
-		M.apply_effect(toxdamage, TOX)
+		var/toxdamage = (toxpwr - (toxpwr*M.getarmor(null, "rad")))

```

Comment:
this still goes negative.

let's take a standard radiation armour of 25.
and 1000 (made up) toxpwr

```var/toxdamage = (1000 - (1000*25)) = -24,000```